operations/fhir_resources_audit/main.py

```diff
@@ -0,0 +1,263 @@
+        """Fetches DocumentRefence ID for the source CSV file as well as BQ table"""
+
+        self.FHIR_URL = Util.get_hde_fhir_url(self.HDE_PREFIX, self.HDE_ENV, self.FHIR_STORE_LOC)
+        docref_url = Util.get_docref_resource_path_url(self.FHIR_URL, self.GCS_FILE_PATH)
+        resources = self.AUTH_QUERY.executeQuery(next_page_url=docref_url)
+        total_matched_resources = resources["total"]
+        for i in range(total_matched_resources):
+            csv_file_info = {}
+            csv_file_info['file'] = resources["entry"][i]['resource']['content'][0]['attachment']['url']
+            csv_file_info['resource_full_url'] = resources["entry"][i]['fullUrl']
+            csv_file_info['id'] = resources["entry"][i]['resource']['id']
+            csv_file_info['meta'] = resources["entry"][i]['resource']['meta']
+
+            #Add details of csv file ingested into BQ table
+            bq_table_info = {}
+            csv_file_docref_id = f"DocumentReference/{resources['entry'][i]['resource']['id']}"
+            docref_url_for_csv = Util.get_bq_table_docref_info(self.FHIR_URL, csv_file_docref_id)
+            resources = self.AUTH_QUERY.executeQuery(next_page_url=docref_url_for_csv)
+            table_info = {}
+            table_info['bq_tbl_docref_id'] = resources['entry'][0]['resource']['target'][0]['reference']
+            table_info['ingested_by'] = resources['entry'][0]['resource']['agent'][0]['who']
+            docref_url_for_bq = Util.get_docref_resources_url(self.FHIR_URL, str(table_info['bq_tbl_docref_id']).split("/")[-1])
+            resources = self.AUTH_QUERY.executeQuery(next_page_url=docref_url_for_bq)
+            table_info['tbl_name'] = resources['entry'][0]['resource']['content'][0]['attachment']['url']
+            table_info['id'] = resources["entry"][0]['resource']['id']
+            csv_file_info['ingested_tbl_info'] = table_info
+
+            self.results.append(csv_file_info)
+
+    def get_reconcilated_resources(self, resource_ref):
+        """Fetches how given FHIR resource reference ID reconciled into and by which pipeline"""
+
+        provenance_recon_url = Util.get_reconcilated_resources_url(self.FHIR_URL, resource_ref)
+        resources = self.AUTH_QUERY.executeQuery(next_page_url=provenance_recon_url)
+        recon_results={}
+        recon_results['reconciled_resource'] = resources['entry'][0]['resource']['target']
+        recon_results['reconciled_by'] = resources['entry'][0]['resource']['agent'][0]['who']
+        return recon_results
+
+
+    def get_transformed_fhir_fhir_resources(self):
+        """Fetches how source records/data transformed into IFS FHIR resources and by which pipeline"""
+
+        for index, source_file_info in enumerate(self.results):
+            ifs_resources = {}
+            provenance_url = Util.get_provenance_resource_path_url(self.FHIR_URL, source_file_info['ingested_tbl_info']['id'])
+            print(provenance_url)
```

Comment:
used for debugging? remove this statement?

Reply:
Done